File: test/bench.py
#!/usr/bin/env python


import numpy as np


def Test1(N):
    A = np.zeros([N,N])
    for i in range(50):
        B = np.sin(A) + np.cos(A)


if __name__ == "__main__":
    from timeit import Timer
    t = Timer("Test1(512)", "from __main__ import Test1")
    print "numpy: test took %f seconds" % t.timeit(number=1)

File: test/bench.lua


require 'lunum'


local function Test1(N)

   local start = os.clock()

   local A = lunum.zeros({N,N})
   for i=0,50 do
      local B = lunum.sin(A) + lunum.cos(A)
   end

   print(string.format("lunum: test took %f seconds", os.clock() - start))
end

collectgarbage("setpause", 50)
collectgarbage("setstepmul", 200)
Test1(512)


-- v0.5.1, no GC: 229
-- v0.5.1, w/ GC: 232

-- v0.5.2, no GC: 201
-- v0.5.2, w/ GC: 197

